### PR TITLE
[backend/frontend] Last fixes on import

### PR DESCRIPTION
**Pull request by Dimfacion**
### Proposed changes

* Fix on import of relative time
* Better readibility on log message/injects when testing a mapper
* Comprehensive error message in case of critical error during import

### Related issues

*
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts [hunk 1]
```diff
@@ -32,10 +35,14 @@ const injectContextForScenario = (scenario: ScenarioStore) => {
       return dispatch(deleteInjectScenario(scenario.scenario_id, injectId));
     },
     onImportInjectFromXls(importId: string, input: InjectsImportInput): Promise<void> {
-      return importXls(scenario.scenario_id, importId, input).then((_value) => new Promise((resolve, _reject) => {
+      return importXls(scenario.scenario_id, importId, input).then((response) => new Promise((resolve, _reject) => {
         dispatch(fetchScenarioInjects(scenario.scenario_id));
         dispatch(fetchScenario(scenario.scenario_id));
         dispatch(fetchScenarioTeams(scenario.scenario_id));
+        const criticalMessages = response.data.import_message.filter((value: ImportMessage) => value.message_level === 'CRITICAL');
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts:42]: Should be put in scenario-action to show success or error depending on result

File: openbas-front/src/utils/Environment.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { useLocation } from 'react-router-dom';
 const MESSENGER$ = new Subject().pipe(debounce(() => timer(500)));
 export const MESSAGING$ = {
   messages: MESSENGER$,
-  notifyError: (text) => MESSENGER$.next([{ type: 'error', text }]),
+  notifyError: (text, sticky) => MESSENGER$.next([{ type: 'error', text, sticky }]),
```
Comment (RomuDeuxfois) [inline openbas-front/src/utils/Environment.js:11]: Should we have a default value for sticky ?
Or it's already handle in the class ?

Comment (Dimfacion) [reply openbas-front/src/utils/Environment.js:11]: As of now, we're using the fact that sticky is undefined and therefore falsy. That would be better to have a default value tho, so I'm changing it.

Comment (Dimfacion) [reply openbas-front/src/utils/Environment.js]: Done

File: openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts (hunk 1 above)
Comment (Dimfacion) [reply openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts]: Should be better now (although I'm unsure how clean the part in openbas-front/src/admin/components/common/injects/ImportUploaderInjectFromXls.tsx is)

**Review by RomuDeuxfois — APPROVED** [review]

File: openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts (hunk 1 above)
Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts]: I don't understand what your mean

**Comment by RomuDeuxfois** [comment]
I just found out with the doc
But the menu names are xls formatters, it should be xls mappers instead, right?
image

**Comment by Dimfacion** [comment]
@RomuDeuxfois Good call, I made the change

**Pull request merged**